core/src/main/java/io/github/tcdl/ChannelManager.java

```diff
@@ -52,15 +52,12 @@ public Producer findOrCreateProducer(final String topic) {
         return producer;
     }
 
-    public synchronized void subscribe(final String topic, final Consumer.Subscriber subscriber) {
-        Consumer consumer = findOrCreateConsumer(topic);
-        consumer.subscribe(subscriber);
-    }
-
-    private Consumer findOrCreateConsumer(final String topic) {
+    public Consumer subscribe(String topic, Subscriber subscriber) {
         Validate.notNull(topic, "field 'topic' is null");
+        Validate.notNull(subscriber, "field 'subscriber' is null");
         Consumer consumer = consumersByTopic.computeIfAbsent(topic, key -> {
```

Comment:
not this is ok, e.g we have few collectors that try to concurrently consume from topic. First one will win and creaae one. Others just need this ONE. Only interest is you get this ONE